Yank/yank.py

```diff
@@ -346,48 +346,35 @@ def _create_phase(self, thermodynamic_state, alchemical_phase):
                 reference_system.addForce(barostat)
 
-        # Check the dimensions of positions.
-        for index in range(len(positions)):
-            n_atoms, _ = (positions[index] / positions[index].unit).shape
-            if n_atoms != reference_system.getNumParticles():
-                err_msg = "Phase {}: number of atoms in positions {} and and " \
-                          "reference system differ ({} and {} respectively)"
-                err_msg.format(alchemical_phase.name, index, n_atoms,
-                               reference_system.getNumParticles())
-                logger.error(err_msg)
-                raise RuntimeError(err_msg)
-
-        # Inizialize metadata storage.
-        metadata = dict()
-
         # Store a serialized copy of the reference system.
         metadata['reference_system'] = openmm.XmlSerializer.serialize(reference_system)
         metadata['topology'] = utils.serialize_topology(alchemical_phase.reference_topology)
 
-        # TODO: Use more general approach to determine whether system is periodic.
-        is_periodic = self._is_periodic(reference_system)
-        is_complex_explicit = len(atom_indices['receptor']) > 0 and is_periodic
-        is_complex_implicit = len(atom_indices['receptor']) > 0 and not is_periodic
-
-        # Make sure pressure is None if not periodic.
-        if not is_periodic: thermodynamic_state.pressure = None
-
         # Create a copy of the system for which the fully-interacting energy is to be computed.
         # For explicit solvent calculations, an enlarged cutoff is used to account for the anisotropic dispersion correction.
         fully_interacting_system = copy.deepcopy(reference_system)
         if is_periodic:
-            # Expand cutoff to maximum allowed
-            # TODO: Should we warn if cutoff can't be extended enough?
-            # TODO: Should we extend to some minimum cutoff rather than the maximum allowed?
+            # Determine minimum box side dimension
             box_vectors = fully_interacting_system.getDefaultPeriodicBoxVectors()
-            max_allowed_cutoff = 0.499 * max([ max(vector) for vector in box_vectors ]) # TODO: Correct this for non-rectangular boxes
+            min_box_dimension = min([max(vector) for vector in box_vectors])
+
+            # Expand cutoff to minimize artifact and verify that box is big enough.
+            # If we use a barostat we leave more room for volume fluctuations or
+            # we risk fatal errors. If we don't use a barostat, OpenMM will raise
+            # the appropriate exception on context creation.
+            # TODO: Make max_allowed_cutoff an option
+            max_allowed_cutoff = 16 * unit.angstroms
+            if thermodynamic_state.pressure and min_box_dimension < 2.25 * max_allowed_cutoff:
+                raise RuntimeError('Barostated box sides must be at least 36 Angstroms '
```

Comment:
This warning should probably compute the `2.25 * max_allowed_cutoff` value and warn about that so we don't forget to change it when we make `max_allowed_cutoff` an option.